yang-semver/draft-ietf-netmod-yang-semver.txt

```diff
@@ -772,75 +809,115 @@ Internet-Draft                 YANG Semver                  October 2023
+   extension for controlling import by YANG semantic version, as well as
+   the rules for how imports are resolved.
 
+5.1.  The recommended-min-version Extension
 
+   The ietf-yang-semver module defines a "recommended-min-version"
+   extension -- a substatement to the "import" statement -- that takes a
+   YANG semantic version as its argument and specifies that the minimum
+   version of the associated module being imported SHOULD be greater
+   than or equal to the specified value.  The specific conditions for
+   determining if a module's version is greater than or equal is defined
+   in Section 5.2 below.  Multiple recommended-min-version statements
+   MAY be specified.  If there are multiple recommended-min-version
+   statements, they are treated as a logical OR.  Removing recommended-
+   min-version statements is considered a backwards compatible change.
+   An example use is:
 
 
 
-Clarke, et al.            Expires 4 April 2024                 [Page 14]
-
-Internet-Draft                 YANG Semver                  October 2023
 
 
-4.  Import Module by Semantic Version
+Clarke, et al.          Expires 13 September 2024              [Page 15]
+
+Internet-Draft                 YANG Semver                    March 2024
 
-   [I-D.ietf-netmod-yang-module-versioning] allows for imports to be
-   done based on a module or a derived revision of a module.  The
-   rev:recommended-min statement can specify either a revision date or a
-   revision label.  The YANG Semver revision-label value can be used as
-   the argument to rev:recommended-min .  When used as such, any module
-   that contains exactly the same YANG semantic version in its revision
-   history may be used to satisfy the import requirement.  For example:
 
            import example-module {
-             rev:recommended-min 3.0.0;
+               ys:recommended-min-version 3.0.0;
            }
 
-   Note: the import lookup does not stop when a non-backward-compatible
-   change is encountered.  That is, if module B imports a module A at or
-   derived from version 2.0.0, resolving that import will pass through a
-   revision of module A with version "2.1.0_non_compatible" in order to
-   determine if the present instance of module A derives from "2.0.0".
+5.2.  Import by YANG Semantic Version Rules
 
-   If an import by recommended-min cannot locate the specified revision-
-   label in a given module's revision history, that import will fail.
-   This is noted in the case of version gaps.  That is, if a module's
-   history includes "1.0.0", "1.1.0", and "1.3.0", an import from
-   recommended-min at "1.2.0" will be unable to locate the specified
-   revision entry and thus the import cannot be satisfied.
+   A module to be imported is considered viable with recommended-min-
+   version if it meets one of the following conditions:
 
-5.  Guidelines for Using Semver During Module Development
+   1.  Has the exact MAJOR, MINOR, PATCH and "_compatible" and
```

Comment:
Overall I like the way you managed to put this together - nice work (I had been pondering a good way to present it).  Should we say "_compatible" or "_non_compatible" (instead of 'and') in bullet 1? (and may be more correct in 2,3,4 though?).